src/Microsoft.DotNet.Wpf/src/PresentationFramework/System/Windows/Controls/RowDefinitionCollectionConverter.cs

```diff
@@ -0,0 +1,117 @@
+        /// <returns>
+        /// The object which was constructoed.
+        /// </returns>
+        /// <exception cref="ArgumentNullException">
+        /// An ArgumentNullException is thrown if the example object is null.
+        /// </exception>
+        /// <exception cref="ArgumentException">
+        /// An ArgumentException is thrown if the object is not null and is not a valid type,
+        /// or if the destinationType isn't one of the valid destination types.
+        /// </exception>
+        /// <param name="context"> The ITypeDescriptorContext for this call. </param>
+        /// <param name="culture"> The CultureInfo which is respected when converting. </param>
+        /// <param name="value"> The Thickness to convert. </param>
+        public override object ConvertFrom(ITypeDescriptorContext context, CultureInfo culture, object value)
+        {
+            if(value != null && value is string input){
+                IProvideValueTarget ipvt = context?.GetService(typeof(IProvideValueTarget)) as IProvideValueTarget;
+                Grid grid = ipvt?.TargetObject as Grid;
+                if(grid != null)
+                {
+                    var collection = new RowDefinitionCollection(grid); // Pass Grid instance
+                    var converter = new GridLengthConverter();
+
+                    TokenizerHelper th = new TokenizerHelper(input, culture);
+                    while(th.NextToken())
+                    {
+                        collection.Add(new RowDefinition { Height = (GridLength)converter.ConvertFromString(th.GetCurrentToken()) });
+                    }
+
+                    return collection;
+                }
+            }            
+            throw GetConvertFromException(value);
+        }
+
+        /// <summary>
+        /// ConvertTo - Attempt to convert a RowDefinitionCollection to the given type
+        /// </summary>
+        /// <returns>
+        /// The object which was constructoed.
+        /// </returns>
+        /// <exception cref="ArgumentNullException">
+        /// An ArgumentNullException is thrown if the example object is null.
+        /// </exception>
+        /// <param name="context"> The ITypeDescriptorContext for this call. </param>
+        /// <param name="culture"> The CultureInfo which is respected when converting. </param>
+        /// <param name="value"> The RowDefintionCollection to convert. </param>
+        /// <param name="destinationType">The type to which to convert the RowDefintionCollection instance. </param>
+        public override object ConvertTo(ITypeDescriptorContext context, CultureInfo culture, object value, Type destinationType)
+        {
+            ArgumentNullException.ThrowIfNull(value);
+            ArgumentNullException.ThrowIfNull(destinationType);
+            if (destinationType == typeof(string) && value is RowDefinitionCollection RowDefinitions)
+            {
+                var parts = new string[RowDefinitions.Count];
```

Comment:
If anything, a `StringBuilder` would do a better job here, though it should be handled on stack / using `ArrayPool<char>` as fallback for longer ones (`DefaultInterpolatedStringHandler` impl for example). Same in `ColumnDefinitions`.

Reply:
Used the `StringBuilder` there